### DISCUSSION TITLE
[@types/level] iterator type is lacking `[Symbol.asyncIterator]()`

### DISCUSSION THREAD

**Discussion by CMCDragonkai** [DefinitelyTyped/DefinitelyTyped#59265] [Issues with a @types package]
The `db.iterator()` which returns `AbstractIterator<any, any>` is capable of being iterated with a `for await`.

Currently this results in an error:

```
Type 'AbstractIterator<any, any>' must have a '[Symbol.asyncIterator]()' method that returns an async iterator.
```

Example:

```
    const i = db.iterator();
    for await (const [k, v] of i) {
      console.log(k.toString(), v.toString());
    }
```

[a comment by an automated account is not shown]

**Comment by CMCDragonkai** [comment]
Could also be affecting the `@types/levelup`, `@types/abstract-leveldown` and `@types/subleveldown`.

**Comment by CMCDragonkai** [comment]
Actually alot of iterator types are no longer correct. See: https://github.com/Level/abstract-leveldown#iterator

Things like `next`, `seek`, `end` are all provided, and `next` and `end` now can return promises.

**Comment by danielfx90** [comment]
Great! Would you like to submit a PR with the changes? I would gladly review it
